src/main/java/Parser.java

```diff
@@ -0,0 +1,121 @@
+import java.time.format.DateTimeParseException;
+import java.util.ArrayDeque;
+
+import javafx.scene.control.Label;
+
+/**
+ * Parser class for CS2103T iP. Handles the logic of the program.
+ */
+public class Parser {
+    private final TaskList list;
+    private final Ui ui;
+
+    /**
+     * Creates a Parser object.
+     * @param list TaskList object for data manipulation.
+     * @param ui Ui object for messages.
+     */
+    public Parser(TaskList list, Ui ui) {
+        this.list = list;
+        this.ui = ui;
+    }
+
+    /**
+     * Main logic of the program.
+     * @return Label with response from program.
+     */
+    public Label parse(String input) {
```

Comment:
Perhaps you should abstract the different commands to their respective functions to avoid the long `parse` method.

Reply:
Else, you should probably use a switch case instead of many else if blocks. Remember to indent correctly if you do go with switch case.